### PR TITLE
feat: update to jina3

### PR DESCRIPTION
**Pull request by Hippopotamus0308**

### CODE REVIEW

File: Dockerfile.gpu [hunk 1]
```diff
@@ -2,7 +2,7 @@ FROM nvidia/cuda:11.2.1-cudnn8-runtime
 
 RUN apt update && DEBIAN_FRONTEND="noninteractive" apt install -y \
     python3.8 python3-pip libsndfile-dev
-RUN JINA_PIP_INSTALL_PERF=1 pip install --no-cache-dir jina~=2.1
+RUN JINA_PIP_INSTALL_PERF=1 pip install --no-cache-dir jina~=3.0.3
```
Comment (nan-wang) [inline Dockerfile.gpu:5]:
```suggestion
RUN JINA_PIP_INSTALL_PERF=1 pip install --no-cache-dir jina~=3.0
```

File: executor/vggish_audio_encoder.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def __init__(
         load_input_from: str = 'uri',
         min_duration: int = 10,
         device: str = '/CPU:0',
-        traversal_paths: Optional[str] = None,
+        traversal_paths: str = None,
```
Comment (nan-wang) [inline executor/vggish_audio_encoder.py:35]:
```suggestion
        traversal_paths: str = '@r',
```

File: executor/vggish_audio_encoder.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def __init__(
         """
 
         super().__init__(*args, **kwargs)
-        self.traversal_paths = traversal_paths or 'r'
+        self.traversal_paths = traversal_paths or '@r'
```
Comment (nan-wang) [inline executor/vggish_audio_encoder.py:57]:
```suggestion
        self.traversal_paths = traversal_paths
```

File: executor/vggish_audio_encoder.py [hunk 4]
```diff
@@ -204,7 +202,7 @@ def _get_input_feature(self, batch_docs):
                 blob_shape_list.append(blob.shape[0])
                 mel_list.append(blob)
         elif self._input == 'log_mel':
-            _mel_list = batch_docs.get_attributes('blob')
+            _mel_list = batch_docs.blobs
```
Comment (nan-wang) [inline executor/vggish_audio_encoder.py:205]: `blob` is renamed to `tensor`. https://docs.jina.ai/get-started/migrate/#common-errors-and-solutions
```suggestion
            _mel_list = batch_docs.tensors
```

File: tests/unit/test_exec.py [hunk 5]
```diff
@@ -92,23 +92,26 @@ def test_encode_gpu(audio_sample_rate):
...
     "traversal_paths, counts",
     [
-        ['c', (('r', 0), ('c', 3), ('cc', 0))],
-        ['cc', (('r', 0), ('c', 0), ('cc', 2))],
-        ['r', (('r', 1), ('c', 0), ('cc', 0))],
-        [['cc', 'r'], (('r', 1), ('c', 0), ('cc', 2))],
+        ['@c', (('@r', 0), ('@c', 3), ('@cc', 0))],
+        ['@cc', (('@r', 0), ('@c', 0), ('@cc', 2))],
+        ['@r', (('@r', 1), ('@c', 0), ('@cc', 0))],
+        ['@cc, r', (('@r', 1), ('@c', 0), ('@cc', 2))],
```
Comment (nan-wang) [inline tests/unit/test_exec.py:98]:
```suggestion
        ['@cc,r', (('@r', 1), ('@c', 0), ('@cc', 2))],
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Review by samsja — APPROVED** [review]

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**